lib/galaxy/datatypes/data.py

```diff
@@ -802,10 +802,13 @@ def count_data_lines(self, dataset):
         """
         data_lines = 0
         with compression_utils.get_fileobj(dataset.file_name) as in_file:
-            for line in in_file:
-                line = line.strip()
-                if line and not line.startswith('#'):
-                    data_lines += 1
+            try:
```

Comment:
Is that an encoding issue ? I.e if you upload something that doesn't have a utf-8 equivalent ?
The change is fine with me, but if that's the case can you add a comment so that we can take this out if / when we have found a better fix ?

Reply:
Yes, this is essentially the problem, with the specific issue I encountered being unable to determine how to split the file data into lines. 

Also agree that we should have a better fix. Would be great to be able to log the issue, and not have the set_meta process cause a job failure. Will add fixme note.